lib/spack/spack/spec.py

```diff
@@ -4553,7 +4557,7 @@ def format_path(
             output_path_components = []
             input_path_components = list(format_string_as_path.parts)
         output_path_components += [
-            fs.polite_filename(self.format(x)) for x in input_path_components
+            disallowed_pattern.sub("_", self.format(x)) for x in input_path_components
```

Comment:
This effectively assumes that `+` can be a part of any file name; if we go that route (vs. having module-specific regexes), then these edits should go into `polite_filename` (i.e. the regex for that should be updated).

However, note this has to work on all systems, and Windows doesn't support `+` in filenames, so `_polite_antipattern` would need to be conditioned for windows/not-windows.

Reply:
the character `+` doesn't occur in the official windows documentation regarding file names: https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file

Reply:
You're right. Let's just include `+` then in the acceptable set of characters :)

Reply:
yup, that was my original PR, I'll just revert it

Reply:
strangely, `[^A-Za-z0-9_.+-]` threw an error while `[^A-Za-z0-9_+.-]` works

Reply:
I thought maybe `+` had to be escaped but that didn't work, and then `.` would also have to be escaped as well

Reply:
I think we may have disallowed `+` because it could be seen as an operator when copied into a Makefile. But I don't think that's gonna be problematic, cause the typical usecase would be

```make
prefix = /hello+world
$(info $(prefix))
```
and that works fine.

Reply:
I can't reproduce it anymore, I must have been doing something else wrong at the time